flink-table/flink-table-planner/src/main/java/org/apache/flink/table/sqlexec/SqlExecutableStatements.java

```diff
@@ -18,62 +18,197 @@
  */
 public class SqlExecutableStatements implements ReflectiveVisitor {
-	private TableEnvironment tableEnv;
+	private FlinkPlannerImpl flinkPlanner;
 
-	private final ReflectUtil.MethodDispatcher<Void> dispatcher =
-		ReflectUtil.createMethodDispatcher(Void.class,
+	private final ReflectUtil.MethodDispatcher<Operation> dispatcher =
+		ReflectUtil.createMethodDispatcher(Operation.class,
 			this,
 			"execute",
 			SqlNode.class);
 
 	//~ Constructors -----------------------------------------------------------
 
-	private SqlExecutableStatements(TableEnvironment tableEnvironment) {
-		this.tableEnv = tableEnvironment;
+	private SqlExecutableStatements(FlinkPlannerImpl flinkPlanner) {
+		this.flinkPlanner = flinkPlanner;
 	}
 
 	/**
-	 * This is the main entrance of executing all kinds of DDL/DML {@code SqlNode}s, different
+	 * This is the main entrance for executing all kinds of DDL/DML {@code SqlNode}s, different
 	 * SqlNode will have it's implementation in the #execute(type) method whose 'type' argument
 	 * is subclass of {@code SqlNode}.
 	 *
 	 * <p>Caution that the {@link #execute(SqlNode)} should never expect to be invoked.
 	 *
-	 * @param tableEnvironment TableEnvironment to interact with
+	 * @param flinkPlanner     FlinkPlannerImpl to convert sql node to rel node
 	 * @param sqlNode          SqlNode to execute on
 	 */
-	public static void executeSqlNode(TableEnvironment tableEnvironment, SqlNode sqlNode) {
-		SqlExecutableStatements statement = new SqlExecutableStatements(tableEnvironment);
-		statement.dispatcher.invoke(sqlNode);
+	public static Operation executeSqlNode(FlinkPlannerImpl flinkPlanner, SqlNode sqlNode) {
+		// validate the query
+		final SqlNode validated = flinkPlanner.validate(sqlNode);
+		SqlExecutableStatements statement = new SqlExecutableStatements(flinkPlanner);
+		try {
+			return statement.dispatcher.invoke(validated);
+		} catch (RuntimeException r) {
+			// try to make the throws more friendly.
+			if (r.getCause() instanceof InvocationTargetException) {
+				final Throwable throwable = r.getCause().getCause();
+				if (throwable instanceof ValidationException) {
+					throw new ValidationException(throwable.getMessage());
+				} else if (throwable instanceof AssertionError) {
+					throw new AssertionError(throwable.getMessage());
+				} else {
+					throw new RuntimeException(throwable);
+				}
+			} else {
+				throw r;
+			}
+		}
 	}
 
 	/**
 	 * Execute the {@link SqlCreateTable} node.
 	 */
-	public void execute(SqlCreateTable sqlCreateTable) {
-		// need to implement.
+	public Operation execute(SqlCreateTable sqlCreateTable) {
+		// set with properties
+		SqlNodeList propertyList = sqlCreateTable.getPropertyList();
+		Map<String, String> properties = new HashMap<>();
+		if (propertyList != null) {
+			propertyList.getList().forEach(p ->
+				properties.put(((SqlProperty) p).getKeyString().toLowerCase(),
+					((SqlProperty) p).getValueString()));
```

Comment:
why toLowerCase for key but not for value?

Reply:
Removed `toLowerCase`